forge-game/src/main/java/forge/game/ability/effects/ConniveEffect.java

```diff
@@ -91,7 +91,6 @@ public void resolve(SpellAbility sa) {
                 }
 
                 CardCollection validCards = CardLists.getValidCards(dPHand, "Card", p, host, sa);
```

Comment:
why was this filter there, after checking for non token, shouldn't all of them be Card ?

Reply:
Or maybe DiscardEffect needs a hard look, too 😆